src/shared/backend/services/flashJobs.ts

```diff
@@ -91,7 +98,56 @@ export const startExecution = async (
       completed: true,
     });
 
-    if (!firmwareData) {
+    if (fetchCloudbuild) {
+      updateStageStatus(jobId, "build", { started: true });
+
+      const params = {
+        release: args.firmware.version,
+        target: args.firmware.target,
+        flags: args.firmware.selectedFlags ?? [],
+      };
+
+      let status = await cloudbuild.createJob(params).catch((e: Error) => {
+        updateStageStatus(jobId, "build", {
+          error: e.message,
+        });
+        return undefined;
+      });
+
+      if (!status || isCancelled(jobId)) {
+        return;
+      }
+
+      // if the build is creating, wait
+      if (status.status !== "BUILD_SUCCESS") {
+        updateStageStatus(jobId, "build", {});
+        status = await cloudbuild
+          .waitForJobSuccess(params, (statusData) => {
+            console.log("Job status updated", statusData);
+            updateStageStatus(jobId, "build", {
+              status: statusData,
+              progress: 0,
+            });
+          })
+          .catch((e: Error) => {
+            updateStageStatus(jobId, "build", {
+              error: e.message,
+            });
+            return undefined;
+          });
+        if (!status || isCancelled(jobId)) {
+          return;
+        }
+      }
+
+      // success
+      firmwareData = await cloudbuild.downloadBinary(
```

Comment:
You should consider leaving in the `Download` step, and then setting the step to `download` here sam as below